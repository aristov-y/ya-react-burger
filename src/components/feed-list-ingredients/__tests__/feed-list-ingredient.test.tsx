import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import FeedListIngredient from '../feed-list-ingredient';
import { useSelector } from 'react-redux';

const store = {
  ingredients: {
    ingredients: [{
      _id: '1',
      name: 'Name',
      image: 'ImageSource'
    }]
  }
}

beforeEach(() => {
  (useSelector as jest.Mock).mockImplementation(fn => fn(store));
});

describe('FeedListIngredient', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<FeedListIngredient id={"1"} />);
    });

    expect(component.toJSON()).toMatchSnapshot();

    act(() => {
      component.update(<FeedListIngredient id={"2"} />);
    });

    expect(component.toJSON()).toMatchSnapshot();

    act(() => {
      component.update(<FeedListIngredient id={"1"} zIndex={2} />);
    });

    expect(component.toJSON()).toMatchSnapshot();

    act(() => {
      component.update(<FeedListIngredient id={"1"} zIndex={2} overLength={3} />);
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
