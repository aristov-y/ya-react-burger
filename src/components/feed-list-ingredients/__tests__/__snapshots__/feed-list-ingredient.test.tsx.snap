// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedListIngredient should render 1`] = `
<div
  className="Ingredient"
>
  <div
    className="IngredientImage"
    style={
      Object {
        "zIndex": 0,
      }
    }
  >
    <img
      alt="Name"
      src="ImageSource"
    />
  </div>
</div>
`;

exports[`FeedListIngredient should render 2`] = `null`;

exports[`FeedListIngredient should render 3`] = `
<div
  className="Ingredient"
>
  <div
    className="IngredientImage"
    style={
      Object {
        "zIndex": 2,
      }
    }
  >
    <img
      alt="Name"
      src="ImageSource"
    />
  </div>
</div>
`;

exports[`FeedListIngredient should render 4`] = `
<div
  className="Ingredient"
>
  <div
    className="IngredientImage"
    style={
      Object {
        "zIndex": 2,
      }
    }
  >
    <img
      alt="Name"
      src="ImageSource"
    />
    <span
      className="text_type_digits-default OverLength"
    >
      +
      3
    </span>
  </div>
</div>
`;
